runtime/worker.rs

```diff
@@ -57,6 +57,24 @@ impl ExitCode {
     self.0.store(code, Relaxed);
   }
 }
+
+// TODO(nayeemrmn): This should be done per-realm once we support `ShadowRealm`,
+// move it to something like `RuntimeOptions::realm_init_cb`.
+pub fn init_runtime_module_map(js_runtime: &mut JsRuntime) {
+  let handles = deno_node::SUPPORTED_BUILTIN_NODE_MODULES
+    .iter()
+    .map(|p| (p, js_runtime.get_module_handle(p.ext_specifier).unwrap()))
+    .collect::<Vec<_>>();
+  js_runtime.clear_module_map();
+  for (p, handle) in handles {
+    js_runtime.inject_module_handle(
+      deno_core::FastString::from_static(p.specifier),
+      deno_core::ModuleType::JavaScript,
+      handle,
+    )
+  }
+}
```

Comment:
@bartlomieju This is how we make extension modules non-importable except for aliasing and exposing `ext:deno_node/` as `node:` specifiers. It's done in `deno_runtime`.

When realm-specific module maps are supported (#15760), we would do this for every realm with a callback and it would look cleaner overall:
```rust
JsRuntime::new(RuntimeOptions {
  // ...
  realm_init_cb: |js_realm| {
    let handles = deno_node::SUPPORTED_BUILTIN_NODE_MODULES
      .iter()
      .map(|p| (p, js_realm.get_module_handle(p.ext_specifier).unwrap()))
      .collect::<Vec<_>>();
    js_realm.clear_module_map();
    for (p, handle) in handles {
      js_realm.inject_module_handle(
        deno_core::FastString::from_static(p.specifier),
        deno_core::ModuleType::JavaScript,
        handle,
      )
    }
  }
})
```

Reply:
This is quite an elegant solution, but I don't like that it's in `runtime/worker.rs`. How about we make it an API in `JsRuntime` that takes a `Vec<(String, String)>` of `(specifier, ext_specifier)` and does this mapping internally.

JsRuntime::inject_module_handle` seems a bit too low-level to expose it (for the time being at least)

Reply:
> How about we make it an API in `JsRuntime` that takes a `Vec<(String, String)>` of `(specifier, ext_specifier)` and does this mapping internally.

The behaviour we want is so specific I can't begin to imagine what this API would be called. IMO, that API and others we've considered are obfuscations of this explicit approach which expose more-or-less the same capability.

Reply:
`JsRuntime::alias_known_modules` would work. Sure, it's specific, but I think it's better than exposing so low level API that allows you to manipulate module handles.